### PR TITLE
Added a way to reset dataset data at the time of new upload

### PR DESCRIPTION
**Pull request by goyal1092**
## Description
Added a way so we can reset dataset data in upload load data.
It's better to do this delete in loaddata because it's an atomic transaction where data is created so it will only delete data if newly uploaded is added to dataset data without any error.

## Related Issue
#238 

## How to test it locally
Reupload same file for a dataset from postman and check if there is duplicate data in indicator data.
Second way to test is if reset is set to true than Django Q logger will have a log statement of : `Deleting previously uploaded data for this dataset`

This statement will not be printed when data is uploaded via admin.
and will also have duplicate indicator data.

## Changelog
Added reset key to loaddata and process_uploaded_file task so we can send reset True or False from async_task to task function.

## Checklist

- [x]  🚀 is the code ready to be merged and go live?
- [x]  🛠 does it work (build) locally

### Pull Request

- [x]  📰 good title
- [x]  📝good description
- [x]  🔖 issue linked
- [x]  📖 changelog filled out

### Commits

- [x]  commits are clean
- [x]  commit messages are clean

### Code Quality

- [x]  🚧 no commented out code
- [x]  🖨 no unnecessary logging
- [x]  🎱 no magic numbers
- [ ]  black was run locally (as part of the pre-commit hook)

### Testing

- [ ]  ✅ added (appropriate) unit tests
- [ ]  💢 edge cases in tests were considered
- [ ]  ✅ ran tests locally & are passing

### CODE REVIEW

File: wazimap_ng/datasets/dataloader.py [hunk 1]
```diff
@@ -32,12 +32,16 @@ def create_groups(dataset, group_names):
 
 
 @transaction.atomic
-def loaddata(dataset, iterable, row_number):
+def loaddata(dataset, iterable, row_number, reset=False):
```
Comment (milafrerichs) [inline wazimap_ng/datasets/dataloader.py:35]: nitpicking here, but can we call it `overwrite` ?

File: wazimap_ng/datasets/tasks/process_uploaded_file.py [hunk 2]
```diff
@@ -12,13 +12,13 @@
...
-def process_file_data(df, dataset, row_number):
+def process_file_data(df, dataset, row_number, reset):
     df = df.applymap(lambda s:s.strip() if type(s) == str else s)
     datasource = (dict(d[1]) for d in df.iterrows())
-    return loaddata(dataset, datasource, row_number)
+    return loaddata(dataset, datasource, row_number, reset)
 
 
-def process_csv(dataset, buffer, chunksize=1000000):
+def process_csv(dataset, buffer, reset, chunksize=1000000):
```
Comment (milafrerichs) [inline wazimap_ng/datasets/tasks/process_uploaded_file.py:21]: let's add a default here as well: 

`overwrite=False`

File: wazimap_ng/datasets/tasks/process_uploaded_file.py [hunk 3]
```diff
@@ -12,13 +12,13 @@
 logger = logging.getLogger(__name__)
 
 
-def process_file_data(df, dataset, row_number):
+def process_file_data(df, dataset, row_number, reset):
```
Comment (milafrerichs) [inline wazimap_ng/datasets/tasks/process_uploaded_file.py:15]: and the default here as well.

File: wazimap_ng/datasets/views.py [hunk 4]
```diff
@@ -97,7 +97,7 @@ def dataset_upload(request, dataset_id):
     update_indicators = request.POST.get('update', False)
     upload_task = async_task(
         "wazimap_ng.datasets.tasks.process_uploaded_file",
-        datasetfile_obj, dataset,
+        datasetfile_obj, dataset, reset=True,
```
Comment (milafrerichs) [inline wazimap_ng/datasets/views.py:100]: can we add another flag in the data. `overwrite` ? and just do it when the flag is set to true.

**Review by milafrerichs — CHANGES_REQUESTED** [review]
